sc/shared/src/test/scala/sigmastate/utxo/BasicOpsSpecification.scala

```diff
@@ -157,6 +159,50 @@ class BasicOpsSpecification extends CompilerTestingCommons
     )
   }
 
+  property("executeFromVar - SigmaProp") {
+    val script = GT(Height, IntConstant(-1)).toSigmaProp
+    val scriptBytes = ValueSerializer.serialize(script)
+    val customExt = Seq(21.toByte -> ByteArrayConstant(scriptBytes))
+    test("executeFromVar", env, customExt,
+      "executeFromVar[SigmaProp](21)",
+      null,
+      true
+    )
+  }
+
+  property("executeFromVar - Int") {
+    val valueBytes = ValueSerializer.serialize(Plus(IntConstant(2), IntConstant(3)))
+    val customExt = Seq(21.toByte -> ByteArrayConstant(valueBytes))
+    test("executeFromVar", env, customExt,
+      "{ executeFromVar[Int](21) == 5 }",
+      null,
+      true
+    )
+  }
+
+  property("executeFromVar - Coll[Byte]") {
+    val bytes = Slice(ByteArrayConstant(Colls.fromArray(Array.fill(5)(1.toByte))), IntConstant(1), IntConstant(3))
+    val valueBytes = ValueSerializer.serialize(bytes)
+    val customExt = Seq(21.toByte -> ByteArrayConstant(valueBytes))
+    test("executeFromVar", env, customExt,
+      "{val ba = executeFromVar[Coll[Byte]](21); ba.size == 2 }",
+      null,
+      true
+    )
+  }
+
+  // test which is showing impossibility of nested Deserialize*
+  property("executeFromVar - deserialize") {
+    val script = DeserializeContext(21.toByte, SSigmaProp)
+    val scriptBytes = ValueSerializer.serialize(script)
+    val customExt = Seq(21.toByte -> ByteArrayConstant(scriptBytes))
+    an [Exception] should be thrownBy test("executeFromVar", env, customExt,
+      "executeFromVar[SigmaProp](21)",
+      null,
+      true
+    )
+  }
+
```

Comment:
Please also add test to LanguageSpecificationV6. 
DeserializeContext was not covered there before because it was not supported by compiler, and scripts are required there.
Now it can be added.

Reply:
??? There is nothing 6.0 specific  or versioned in general specific in this PR.

Reply:
DeserializeContext in not covered in LanguageSpecificationV6, above I explained why. Now it can be covered, but also probably was not used in practice due to lack of ErgoScript support. Now it will be available, so better be covered in the specification.

Reply:
Sorry, I do not understand, what do you propose to test ? I mean, which `DeserializeContext` properties, and why in LanguageSpecificationV6 specifically. All other tests in this specification are testing versioned behavior of features added in 6.0. And ScalaDoc for `LanguageSpecificationBase` is also proposing to test versioned behavior in `LanguageSpecification` tests. Have you changed ScalaDoc in some PR maybe ?

Reply:
There is no contradiction, every feature can be tested for many versions, but at least for one. Importantly, all components of the interpreter is tested, parsing, compiler stages, tree serialization, evaluation on different inputs, positive and negative cases, expected values are checked. That is why there should be tests for each language feature (operation) here. Versioning is not a central part of it, but important capability.

Reply:
Then I guess `DeserializeContext`  should be in LanguageSpecificationV3 along with all features existing since the mainnet launch. But then you need to open an issue at least about doing what you are proposing for all the nodes existing since the launch, to avoid chaotic implementation of 2-3 nodes before it will be forgotten forever. And the question about properties is still  not answered. A typical check in LSV6 is like 

```
      lazy val bitAnd = newFeature(
      { (x: (Int, Int)) => x._1 & x._2 },
      "{ (x: (Int, Int)) => x._1 & x._2 }",
      sinceVersion = VersionContext.V6SoftForkVersion)
      forAll { x: Int =>
        Seq(toAbs).foreach(_.checkEquality(x))
      }
      forAll { x: (Int, Int) =>
        Seq(compareTo, bitOr, bitAnd).foreach(_.checkEquality(x))
      }
```

So a check is testing that for V6 a new features has some behavior. What do you want to check about `DeserializeContext` in LSV* ?

Reply:
Yes, it is an existing feature so can be added to LSV5 as existingFeature(...).
I believe described above what I "want to check" by creating a new `property(){...}` for  DeserializeContext.
Most of the other cases like this are either not known or already recorded as issues. Don't think this PR depend on this.

Reply:
Trying to implement it , I think, there is inherent issue in `SigmaDslTesting` still, talking about DeserializeContext node, that is, the DSL does not have a function corresponding to executeFromVar and the node, e.g. I dont see how anything but trivial substitutions with evaluated data (via `getVar` can be done), for example, the following test in BasicOpsSpecification:

```
    val bytes = Slice(ByteArrayConstant(Colls.fromArray(Array.fill(5)(1.toByte))), IntConstant(1), IntConstant(3))
    val valueBytes = ValueSerializer.serialize(bytes)
    val customExt = Seq(21.toByte -> ByteArrayConstant(valueBytes))
    test("executeFromVar", env, customExt,
      "{val ba = executeFromVar[Coll[Byte]](21); ba.size == 2 }",
     ... 
```

does not have an alternative due to that